spec/controllers/api/v1/users_controller_spec.rb

```diff
@@ -0,0 +1,130 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+require 'json'
+
+RSpec.describe Api::V1::UsersController, type: :controller do
+  fixtures :users
+  let(:user) { users(:one) }
+
+  describe 'GET #show' do
+    context 'when is succesful' do
+      before { get :show, params: { id: user.id } }
+
+      it 'return succesful' do
+        expect(response).to have_http_status(:success)
+      end
+
+      it 'returns the user' do
+        # Test to ensure response contains the correct email
+        expect(JSON.parse(response.body)).to include('email' => user.email)
+      end
+    end
+
+    context 'when is not succesful' do
+      before { get :show, params: { id: 0 } }
+
+      it 'returns not found' do
+        expect(response).to have_http_status(:not_found)
+      end
+
+      it 'returns the error message' do
+        expect(response.body).to include('User not found')
+      end
+    end
+  end
+
+  describe 'POST #create' do
+    def create_call(user_params)
+      post :create, params: { user: user_params }
+    end
+
+    context 'when is succesful' do
+      let(:new_user) { { email: 'test@test.org', password: '123456' } }
+
+      before { create_call(new_user) }
+
+      it 'returns successful' do
+        expect(response).to have_http_status(:created)
+      end
+
+      it 'returns the user' do
+        expect(JSON.parse(response.body)).to include('email' => new_user[:email])
+      end
+    end
+
+    context 'when is not succesful' do
+      let(:new_user) { { email: 'test.org', password: '123456' } }
+
+      before { create_call(new_user) }
+
+      it 'returns code 422' do
+        expect(response).to have_http_status(:unprocessable_entity)
+      end
+
+      it 'returns a error' do
+        expect(JSON.parse(response.body)).to include('email' => ['is invalid'])
+      end
+    end
+  end
+
+  describe 'PUT #update' do
```

Comment:
What happen if the update fail, for example if the email is wrong, we need to add a test for it

Reply:
The test for this case is `'when params are invalid'`, passing `'bad_email'` as email parameter. I could change the context to be more consistent with the rest.